src/App.js

```diff
@@ -10,27 +9,77 @@ class App extends Component {
   constructor() {
   super()
     this.state = {
-      movies: movieData.movies,
-      clicked: false
+      movies: [],
+      movieSummary: null,
+      clicked: false,
+      loading: false,
+      movieSummaryLoading: false,
+      errorMessage: null,
+      error: false
     }
   }
   
-  //stuff
-  displayMovieSummary = (id) => {
-    const selectedMovie = this.state.movies.filter(movie => movie.id === id) 
-      this.setState({movies: selectedMovie, clicked: true});
-  }
+ 
+componentDidMount = () => {
+    this.setState({
+      loading: true
+    })
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
+      .then(res => res.json())
+      .then(movies => {
+        this.setState({movies: movies.movies})
+      })
+      .catch(error => {
+        this.setState({
+          errorMessage: error.message,
+          error: true
+        })
+      })
+      .finally(() => {
+        this.setState({
+          loading: false
+        })
```

Comment:
Instead of using `.finally()`, you can set loading to false on line 30.  Something like `this.setState({ movies: movies.movies, loading: false });`